generatorcore/industry2018/energy_demand.py

```diff
@@ -0,0 +1,114 @@
+# pyright: strict
+from dataclasses import dataclass
+from .co2e import CO2e
+from ..inputs import Inputs
+from .co2e_per_t import CO2e_per_t
+from .co2e_basic import CO2e_basic
+from .co2e_per_mwh import CO2e_per_MWh
+from .co2e_with_pct_energy import CO2e_with_pct_energy
+
+
+@dataclass
+class Production:
```

Comment:
I don't have a good answer here yet, and I'm perfectly happy to merge this as is, and tackle this separately. But I again want to float my thoughts so you can think about this too.  I think the names here aren't exactly right.

Why is the module called `energy_demand`? Or if the name is right why is the class called `Production`? Also if the module name is right maybe we should just call `calc_energy_demand`,`calc` and call it with a fully qualified path.

Maybe my unease about the name comes from the feeling that this module isn't primarily about energy demand but about CO2e caused by energy consumption (as opposed to co2e caused by the process itself). I think. To be honest I'm not sure because if that is the case where do we compute the process based CO2e? And if this module is about both than the name seems even weirder...

Reply:
@bgrundmann, @Ultramarin95: 
Ich stimme dir zu, das Naming ist hier noch nicht ganz glücklich gewählt. 
Vielleich hat Leon hierzu noch ein paar gute Ideen oder kann etwas Infos zu den Abkürzungen geben.

Mein Ziel ist folgendes:
Es gibt ja Sektoren, die Energie bereitstellen und Sektoren die Energie verbrauchen.
Industry ist so ein Sektor, der Energie verbraucht und für diese Sektoren passiert immer sowas in der Art:
1) calc_energy_demand oder calc_CO2e oder ... Namensvorschlag ;)
2) calc_energy_sources bzw. verteile den CO2 Bedarf auf die Energy Sources

zu 1) --> hier kommen member raus die p_... heißen, daher habe ich das irgendwie production genannt
zu 2) --> hier kommen member raus die s_... heißen, daher habe ich das energy_source genannt

@Ultramarin95: Für was steht denn p_... ?

Grundsätzlich würde ich vorschlagen die calc methode für alles Sektoren genauso zu untergliedern. 
Bessere Namensvorschläge nehme ich gern entgegen ;)

Reply:
@Ultramarin95: Bitte um Feedback ;)

Reply:
Wird hier weiter bearbeitet: https://github.com/GermanZero-de/localzero-generator-core/pull/214